commonknowledge/meep-intelligence-hub
# feat: add (unstyled) registration page and activation email

## Description
Adds a new front end page /register, to hold the registration form. Also adds an email
template for the activation email.

Also simplifies the front-end somewhat by splitting `useRequireAuth` into `useRequireAuth` and `useRequireNoAuth` for the pages (like login and register) that expect the user to _not_ be logged in.

## Motivation and Context
We need to allow users to register!

## How Can It Be Tested?
1. Check out the branch, start the backend and frontend, then go to localhost:3000/register.
2. Fill in the form and submit it. It should succeed.
3. The email content should be captured by MailHog, visit localhost:8025 to view it.
4. Click the link in the email. You should now be able to log in as the new user.
